engine/api/workflow_queue_artifact.go

```diff
@@ -199,7 +200,7 @@ func (api *API) postWorkflowJobArtifactWithTempURLCallbackHandler() Handler {
 			return sdk.WrapError(sdk.ErrNotFound, "postWorkflowJobArtifactWithTempURLCallbackHandler> Unable to find artifact, key:%s", cacheKey)
 		}
 
-		if art != cachedArt {
+		if !cmp.Equal(art, cachedArt) {
```

Comment:
What was the difference between the object ?
The struct in only composed of comparable types, so we don't need reflection to compare it

Eventually if the comparison is a bit specific implement an Equal function on the type to avoid a new dependency and reflection